[Smart Accounts Kit] Improve redelegation guides (#2665)
MetaMask/metamask-docs: smart-accounts-kit/guides/delegation/create-redelegation.md

</TabItem>
</Tabs>

### Limit redelegation using caveats

> When creating a redelegation, you can also use the [caveats](../../reference/delegation/caveats.md) provided by the toolkit to narrow the authority for the Carol. For example, you can further limit the authority so that the Carol so she can only use the delegation once.

Reviewer: ```suggestion
When you create a redelegation, apply the toolkit's [caveats](../../reference/delegation/caveats.md) to narrow Carol's authority. For example, you can limit the authority so Carol can use the delegation only once.
```

> To apply caveats, create the `Delegation` object and use [`createCaveatBuilder`](../../reference/delegation/index.md#createcaveatbuilder). Use [`getDelegationHashOffchain`](../../reference/delegation/index.md#getdelegationhashoffchain) to get the delegation hash, then provide it as the `authority` field. 

Reviewer: Should we add something here (and the previous example) to specify what makes it single-use? e.g

"In this example the [`limitedCalls`](link) caveat is set `1` and added to the redelegation."
